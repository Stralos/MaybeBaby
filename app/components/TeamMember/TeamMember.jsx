// @flow
import React, { PureComponent } from 'react';
import styled from 'styled-components';

const TeamMemberStyled = styled.div`
  display: flex;
  margin-bottom: 24px;
  &:last-child {
    margin-bottom: 0;
  }
`;
export const Profile = styled.div`
  margin-left: 16px;
  display: flex;
  flex-direction: column;
  justify-content: space-between; 
`;
export const Profession = styled.div`
`;
export const Description = styled.div`
  font-family: "Marcellus SC",serif;
  font-size: 1.2em;
  color: #999;
`;
export const Name = styled.div`
  font-family: "Marcellus SC";
  font-size: 1.75rem;
  font-weight: 500;
  line-height: 1.1;
`;

const INITIALS_SIZE = '120px';
export const Initials = styled.div`
  font-size: 2.25em;
  color: #d9d9d9;
  font-family: "Playfair Display",serif;
  border-radius: 50%;
  height: ${INITIALS_SIZE};
  width: ${INITIALS_SIZE};
  min-width: ${INITIALS_SIZE};
  min-height: ${INITIALS_SIZE};
  background-color: #777;
  line-height: ${INITIALS_SIZE};
  text-align: center;
`;
export const Image = styled.img`
  height: ${INITIALS_SIZE};
  width: ${INITIALS_SIZE};
  border-radius: 50%;
`;

export type Props = {
  name: string,
  initials: string,
  image: ?string,
  profession: ?string,
  description: ?string,
}

class TeamMember extends PureComponent<Props> {
  renderImage = () => {
    const { image, initials } = this.props;
    return image ?
      <Image src={image} /> :
      <Initials>{initials}</Initials>;
  }

  render = () => {
    const {
      name,
      description,
      profession,
    } = this.props;
    return (
      <TeamMemberStyled>
        {this.renderImage()}
        <Profile>
          <Name>
            {name}
          </Name>
          {description && (
            <Description>
              {description}
            </Description>
          )}
          {profession && (
            <Profession>
              {profession}
            </Profession>
          )}
        </Profile>
      </TeamMemberStyled>
    );
  };
}

export default TeamMember;
