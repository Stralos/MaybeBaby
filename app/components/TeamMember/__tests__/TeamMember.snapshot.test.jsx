/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import TeamMember from '../TeamMember';
import type { Props } from '../TeamMember';

describe('TeamMember Component Snapshot', () => {
  let props: Props;

  beforeEach(() => {
    props = {
      name: 'James Smith',
      initials: 'JS',
      image: 'www.james-smith.com/image/me.jpg',
      profession: 'Hair Dresser',
      description: 'Likes to talk a lot',
    };
  });

  it('it should render correctly with an image', () => {
    const tree = renderer.create(
      <TeamMember {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with initials', () => {
    props.image = null;
    const tree = renderer.create(
      <TeamMember {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render description', () => {
    props.description = null;
    const tree = renderer.create(
      <TeamMember {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render profession', () => {
    props.profession = null;
    const tree = renderer.create(
      <TeamMember {...props} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
