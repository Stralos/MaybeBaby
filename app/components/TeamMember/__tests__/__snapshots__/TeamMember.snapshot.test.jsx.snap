// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember Component Snapshot it should render correctly with an image 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  margin-left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-family: "Marcellus SC",serif;
  font-size: 1.2em;
  color: #999;
}

.c2 {
  font-family: "Marcellus SC";
  font-size: 1.75rem;
  font-weight: 500;
  line-height: 1.1;
}

<div
  className="c0"
>
  <img
    className=""
    src="www.james-smith.com/image/me.jpg"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      James Smith
    </div>
    <div
      className="c3"
    >
      Likes to talk a lot
    </div>
    <div
      className=""
    >
      Hair Dresser
    </div>
  </div>
</div>
`;

exports[`TeamMember Component Snapshot should not render description 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  margin-left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  font-family: "Marcellus SC";
  font-size: 1.75rem;
  font-weight: 500;
  line-height: 1.1;
}

<div
  className="c0"
>
  <img
    className=""
    src="www.james-smith.com/image/me.jpg"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      James Smith
    </div>
    <div
      className=""
    >
      Hair Dresser
    </div>
  </div>
</div>
`;

exports[`TeamMember Component Snapshot should not render profession 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  margin-left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-family: "Marcellus SC",serif;
  font-size: 1.2em;
  color: #999;
}

.c2 {
  font-family: "Marcellus SC";
  font-size: 1.75rem;
  font-weight: 500;
  line-height: 1.1;
}

<div
  className="c0"
>
  <img
    className=""
    src="www.james-smith.com/image/me.jpg"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      James Smith
    </div>
    <div
      className="c3"
    >
      Likes to talk a lot
    </div>
  </div>
</div>
`;

exports[`TeamMember Component Snapshot should render correctly with initials 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 24px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c2 {
  margin-left: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-family: "Marcellus SC",serif;
  font-size: 1.2em;
  color: #999;
}

.c3 {
  font-family: "Marcellus SC";
  font-size: 1.75rem;
  font-weight: 500;
  line-height: 1.1;
}

.c1 {
  font-size: 2.25em;
  color: #d9d9d9;
  font-family: "Playfair Display",serif;
  border-radius: 50%;
  height: 120px;
  width: 120px;
  min-width: 120px;
  min-height: 120px;
  background-color: #777;
  line-height: 120px;
  text-align: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    JS
  </div>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      James Smith
    </div>
    <div
      className="c4"
    >
      Likes to talk a lot
    </div>
    <div
      className=""
    >
      Hair Dresser
    </div>
  </div>
</div>
`;
